runtime/lua/vim/tshighlighter.lua

```diff
@@ -56,6 +55,19 @@ function TSHighlighter:set_query(query)
   end
   self.query = query
 
+  self.hl_cache = {}
```

Comment:
I think this can be done better with `__index` metamethod.  It might be more obvious what's happening?

Reply:
You mean adding an `__index` metamethod to the cache so that it fills itself on the fly ?

Reply:
Hopefully I got it right !